### PR TITLE
T responder dao controller

### PR DESCRIPTION
**Pull request by angela81ku**

### CODE REVIEW

File: controllers/PollResponseController.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+     * @return PollResponseController
+     */
+    public static getInstance = (app: Express): PollResponseController => {
+        if(PollResponseController.pollResponseController == null) {
+            PollResponseController.pollResponseController = new PollResponseController();
+
+            //Restful User Web service API
+            app.get('/api/users/:uid/pollsresponded',PollResponseController.pollResponseController.findPollResponsesByUser);
+            app.get('/api/usersrepsonded/polls/:pid',PollResponseController.pollResponseController.findAllUsersReplyPollResponse);
+            app.get('/api/polls/:pid',PollResponseController.pollResponseController.findPollResponseByPollId);
```
Comment (vivekananda-reddy) [inline controllers/PollResponseController.ts]: Do we need this api? This was not part of initial API discussions. If we need this then we need to change the api path as it clashes with other TCreator API

File: controllers/PollResponseController.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+     * database, if the Poll is close, then the response can't be created.
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON containing the new pollResponse that was inserted in the
+     * database
+     */
+    createPollResponse = async (req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+        // let userId = req.session['profile']._id;
+
```
Comment (vivekananda-reddy) [inline controllers/PollResponseController.ts]: We don't need a try catch block her like the one added for updateResponse function? If tested with incorrect pollID and it works then ignore this comment. If not tested please test

File: controllers/PollResponseController.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    /**
+     * @param {Request} req Represents request from client, including path
+     * parameter pid and uid representing pollId and userId, if the poll is already closed,
+     * the deletePollResponse will send a 404 code.
+     *  @param {Response} res Represents response to client, including status
+     *  on whether deleting a pollResponse was successful or not
+     */
+    deletePollResponse = async (req: Request, res: Response) =>{
+        let isPollClosed = await PollResponseController.pollResponseDao.isPollClosed(req.params.pid)
```
Comment (vivekananda-reddy) [inline controllers/PollResponseController.ts:146]: We don't need a try catch block her like the one added for updateResponse function? If tested with incorrect pollID and it works then ignore this comment. If not tested please test

**Review by vivekananda-reddy — CHANGES_REQUESTED** [review]

File: controllers/PollResponseController.ts (hunk 1 above)
Comment (angela81ku) [reply controllers/PollResponseController.ts]: change it to different api

File: controllers/PollResponseController.ts (hunk 3 above)
Comment (angela81ku) [reply controllers/PollResponseController.ts:146]: delete it, but add isPollClosed DAO catch error

File: controllers/PollResponseController.ts (hunk 2 above)
Comment (angela81ku) [reply controllers/PollResponseController.ts]: delete it, but add isPollClosed DAO catch error

**Pull request merged**